azurerm/internal/services/network/firewall_resource.go

```diff
@@ -132,6 +165,42 @@ func resourceArmFirewall() *schema.Resource {
 				},
 			},
 
+			"virtual_hub": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"virtual_hub_id": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validate.ValidateVirtualHubID,
+						},
+						"public_ip_count": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							ValidateFunc: validation.IntAtLeast(1),
+							Default:      1,
+						},
+						"public_ip_addresses": {
+							Type: schema.TypeList,
```

Comment:
@magodo Is it possible to test scenarios involving the `public_ip_addresses` attribute? I'm curious what happens if you scale out the number of public IPs and only specify a subset of them. I also suspect if there are ordering issues, this attribute may need to be a set.

Reply:
@manicminer The reason why this is a list is because it is a Computed attribute, a list is more user friendly. Regarding to test for `public_ip_addresses`, I guess we need to enhance the data source of the `firewall` resource so that we can retrieve the Computed `public_ip_addresses` and feed a subset of them to the resource for another apply?

Reply:
@manicminer I added the test accordingly, please take another look!

Reply:
@magodo thanks, that's good for the scaling in case, but can we also add a test for the scaling out case?

Reply:
@manicminer I meant that I've added the scaling out case in the last commit [here](https://github.com/terraform-providers/terraform-provider-azurerm/pull/8879/commits/b98806e5fad991c30e01c69c88bfb0c21cbd89d7#diff-dcc9ede2bfb4a29b38331b171874632050bfa7d2551d63fa98c04f971049b5c0R347).

Reply:
@magodo Thanks, I saw that. I'd describe that case as scaling in / down. The use case I'm interested in is scaling up / out and how that'll affect the `public_ip_addresses` attribute.

For example, say I have `public_ip_count = 2` and `public_ip_addresses = ["1.2.3.4", "1.2.3.5"]`, and then I increase `public_ip_count = 4` and apply. Are the newly provisioned public IP addresses returned in the `AzureFirewallPropertiesFormat.HubIPAddresses.PublicIPs.Addresses` field? If so, what happens to the Terraform plan?

Reply:
@manicminer Thank you for clarify that (i mistakely regard scaling out as scaling down...).

In this case, the first apply will take effect and then set the `public_ip_addresses` with 4 elements (adding two new ip addresses on top of the other two ips you've specified). The second terraform plan will show diff if the user is literally setting two elements in the config. However, in the scaling out/up use case, the user should not set that property at all, as I mentioned in the [document](https://github.com/terraform-providers/terraform-provider-azurerm/pull/8879/files?file-filters%5B%5D=.markdown#diff-50fbb9bccec1371a75f26bb6531f3339aaf13f78b4d8321fbc3a06284410e7a1R131). This property is mostly meant to be computed, and only useful in scale in/down use case.

Reply:
@magodo Thanks for clarifying, that was my interpretation. In this case, the API is treating the field differently depending on how the value of `public_ip_count` is changing - essentially this should be two separate fields in the API. However, with it being one field having multiple behaviors, we can't really support this. Attributes that are only set for the duration of a single apply are going to produce inconsistent plans in pretty much any change scenario.

I suggest we simply make this field Computed (read-only), which will expose the allocated IPs for reference by the operator in other resources and I think is more important to enable.

Reply:
@manicminer This makes total sense to me! I have made the changes to make `public_ip_addresses` to be computed only in the latest commit, please take another look!